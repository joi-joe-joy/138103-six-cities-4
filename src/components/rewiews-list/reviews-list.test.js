import React from "react";
import renderer from "react-test-renderer";
import ReviewsList from "./reviews-list.jsx";

const reviews = [
  {
    id: 5,
    author: {
      photo: `img/avatar-angelina.jpg`,
      name: `Monica`
    },
    description: `bedrooms â€“ couples and groups of friends will find this accommodation`,
    date: 1593095836000,
    rating: 3
  },
  {
    id: 6,
    author: {
      photo: `img/avatar-angelina.jpg`,
      name: `Bell Gover`
    },
    description: `friends will find this accommodation`,
    date: 1603095836000,
    rating: 5
  }
];

describe(`Render ReviewsList`, () => {
  it(`Render ReviewsList correctly`, () => {
    const tree = renderer
      .create(<ReviewsList reviews={reviews}/>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Render ReviewsList correctly empty`, () => {
    const tree = renderer
      .create(<ReviewsList reviews={[]}/>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
