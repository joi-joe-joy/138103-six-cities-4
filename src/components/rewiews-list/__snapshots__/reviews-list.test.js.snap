// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ReviewsList Render ReviewsList correctly 1`] = `
<ul
  className="reviews__list"
>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Bell Gover"
          className="reviews__avatar user__avatar"
          height="54"
          src="img/avatar-angelina.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Bell Gover
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "100%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        friends will find this accommodation
      </p>
      <time
        className="reviews__time"
        dateTime="2020-10-19"
      >
        October 2020
      </time>
    </div>
  </li>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Monica"
          className="reviews__avatar user__avatar"
          height="54"
          src="img/avatar-angelina.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Monica
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "60%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        bedrooms â€“ couples and groups of friends will find this accommodation
      </p>
      <time
        className="reviews__time"
        dateTime="2020-06-25"
      >
        June 2020
      </time>
    </div>
  </li>
</ul>
`;

exports[`Render ReviewsList Render ReviewsList correctly empty 1`] = `
<ul
  className="reviews__list"
/>
`;
